api/rooms/views/map.py

```diff
@@ -0,0 +1,116 @@
+import json
+from pyproj            import Transformer
+
+from django.db.models  import Q
+from django.views      import View
+from django.http       import JsonResponse
+
+from ..models          import Room, TradeType
+
+def wgs84_to_utmk(longitude, latitude):
+    return Transformer.from_crs("epsg:4326", "epsg:5178").transform(latitude, longitude)
+
+def utmk_to_wgs84(utmk_x, utmk_y):
+    return Transformer.from_crs("epsg:5178", "epsg:4326").transform(utmk_x, utmk_y)
+
+def make_room_filter(func):
+    def wrapper(self, request, *args, **kwargs):
+        filter = json.loads(request.GET.get('filter', '{}'))
+        room_type  = filter.get('roomType', [])
+        trade_type = filter.get('tradeType', [])
+        locations  = json.loads(request.GET.get('location', '[]'))
+        dong_code  = request.GET.get('dongCode')
+        q          = Q()
+
+        if len(locations) == 2 and \
+            len(locations[0]) == 2 and len(locations[1]) == 2:
+            sw = wgs84_to_utmk(locations[0][0], locations[0][1])
+            ne = wgs84_to_utmk(locations[1][0], locations[1][1])
+
+            q &= Q(location__utmk_x__range = [sw[0], ne[0]], 
+                location__utmk_y__range = [sw[1], ne[1]])
+
+        if trade_type:
+            trade_q      = Q()
+            deposit      = filter.get('deposit', [0, 999999999])
+            monthly_rent = filter.get('monthlyRent', [0, 999999999])
+            sale         = filter.get('sale', [0, 999999999])
+            
+            if TradeType.Enum.MONTHLY_RENT.value in trade_type :
+                trade_q |= Q(
+                    trade_type_id=TradeType.Enum.MONTHLY_RENT.value, 
+                    monthly_rent__range=monthly_rent, deposit__range=deposit)
+
+            if TradeType.Enum.DEPOSIT.value in trade_type :
+                trade_q |= Q(trade_type_id=TradeType.Enum.DEPOSIT.value, 
+                    deposit__range=deposit)
+
+            if TradeType.Enum.SALE.value in trade_type :
+                trade_q |= Q(trade_type_id=TradeType.Enum.SALE.value,
+                    sale__range=sale)
+
+            q &= Q(trade_q)
+
+        if room_type:
+            q &= Q(room_type_id__in=room_type)
+
+        if dong_code:
+            q &= Q(location__dong_code=dong_code)
+
+        request.q = q
+        
+        return func(self,request, *args, **kwargs)
+
+    return wrapper
+
+class RoomListView(View):
+    @make_room_filter
+    def get(self, request): 
+        next_page = 2
+        has_more  = False
+        page      = int(request.GET.get('page', 0))
+        limit     = int(request.GET.get('limit', 15))
```

Comment:
`offset` `limit`이 아니라 page로 받는 이유가 있을까요?
```suggestion
        next_page = 2
        has_more  = False
        page      = int(request.GET.get('page', 0))
        limit     = int(request.GET.get('limit', 15))
```

Reply:
`page` 는 몇번째 페이지인지 받을려고 하였고,
`limit` 는 한 페이지에 몇개의 결과가 있을 수 있는지 받을려고 하였는데, 지금 생각해보니 명칭이 모호한것 같습니다.

1. page, page_size 로 갈지
2. offset, limit 갈지 정해야될 것 같습니다.

Reply:
제가의도한 코드는 page, page_size 이므로 limit를 page_size으로 고치고, 추후 프론트분들과 붙여볼 시에 편하신 방식으로 수정하도록 하겠습니다.

Reply:
해당 엔드포인트를 여러 곳에서 호출해야되는데, A화면에서는 100개, B화면에서는 10개, C서비스에서는 1000개를 받아오고 싶다고하면 이때도 페이지 number를 기준으로 받아와야 되는 문제가 생기죠!

pagination은 offset, limit으로 구현해주셔야 flexible하게 사용될 수 있습니다!

Reply:
수정완료하였습니다.